modules/dnn/test/test_int.cpp

```diff
@@ -6,9 +6,54 @@
 // Third party copyrights are property of their respective owners.
 
 #include "test_precomp.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
 
 namespace opencv_test { namespace {
 
+template <class T>
+Mat randomIntMatImpl(const std::vector<int> &sizes, int64_t low, int64_t high, int matType)
+{
+    CV_Assert(low < high);
+    CV_Assert(high - low < RAND_MAX);
+
+    Mat m(sizes, matType);
+
+    int size = 1;
+    for (int s: sizes) {
+        size *= s;
+    }
+
+    T* data = m.ptr<T>();
+    for (int i = 0; i < size; ++i)
+    {
+        data[i] = low + std::rand() % (high - low);
+    }
+
+    return m;
+}
+
+Mat randomIntMat(const std::vector<int> &sizes, int64_t low, int64_t high, int matType)
+{
+    if (matType == CV_32S)
+        return randomIntMatImpl<int32_t>(sizes, low, high, matType);
+    else if (matType == CV_64S)
+        return randomIntMatImpl<int64_t>(sizes, low, high, matType);
+    else
+        CV_Error(Error::BadDepth, "Unsupported type");
+    return Mat();
+}
```

Comment:
Doesn't [randu](https://docs.opencv.org/5.x/d2/de8/group__core__array.html#ga1ba1026dca0807b27057ba6a49d258c0) fit required data types?

Reply:
Does it work with int64? I need big values to check that we don't cast int64 values inside dnn

Reply:
That seems working, yes:
```cpp
Mat a(1, 5, CV_64S);
cv::randu(a, pow(2l, 5), LLONG_MAX);
std::cout << a << std::endl;
```
```
[2598889899411822216, 5139869645836315984, 5791353558368746828, 1845221523294552719, 2245250195598578852]
```

```